Lib/glyphsLib/builder/glyph.py

```diff
@@ -123,7 +123,22 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph):  # noqa: C901
         ufo_glyph.lib[GLYPHLIB_PREFIX + "subCategory"] = subCategory
 
     # load width before background, which is loaded with lib data
-    width = layer.width
+
+    # The width may be taken from another master via the customParameters
+    # 'Link Metrics With Master' or 'Link Metrics With First Master'.
+    master = self.font.masters[layer.associatedMasterId or layer.layerId]
+    metric_source = master.metricsSource.id
+    metric_layer = self.font.glyphs[glyph.name].layers[metric_source]
+    if metric_layer:
+        width = metric_layer.width
+    else:
+        width = None
+    if layer.width != width:
```

Comment:
you're only applying it if width is not None (or if metric_layer is true); the `!=` may catch the case when width is None and layer.width is not None and trigger the logging debug message, when in fact you haven't applied anything because width was None. I think the logger.debug() call should be moved in the `if metric_layer:` branch after you override `width = metric_layer.width`